### ISSUE TITLE
[meta] Ibis backend public API stability

### ISSUE THREAD

**Issue by lostmygithubaccount** [ibis-project/ibis#8994]
**Acceptance criteria**: This quarter, we will have a precise plan and initial work toward a stable public backend interface.

The purpose of this work is to set a stable foundation for new backends to be added, particularly be external contributors/organizations. We should work toward allowing external backends to be fully contributed, maintained, and owned in external repositories.

**Comment by ncclementi** [comment]
After re-triaging/scoping some issues these issues belong to this epic, leaving them here for reference:

## P0
- [ ] https://github.com/ibis-project/ibis/issues/6593  

## P1 
- [ ] https://github.com/ibis-project/ibis/issues/9000 (needs to wait until 9.1) 
- [ ] https://github.com/ibis-project/ibis/issues/8735  (PR open https://github.com/ibis-project/ibis/pull/8863 blocked on #6593)
- [ ] https://github.com/ibis-project/ibis/issues/8382 (PR open https://github.com/ibis-project/ibis/pull/8864 needs review)
- [ ] https://github.com/ibis-project/ibis/issues/8584 
- [ ] https://github.com/ibis-project/ibis/issues/8898 

## P2 
- [ ] https://github.com/ibis-project/ibis/issues/8133 
- [ ] https://github.com/ibis-project/ibis/issues/8727 
- [ ] https://github.com/ibis-project/ibis/issues/8866

**Comment by gforsyth** [comment]
I've opened a PR with a POC for lightly enforcing signature compatibility between backends in #9383

**Comment by ncclementi** [comment]
Closing in favor of https://github.com/ibis-project/ibis/issues/9638
